### ISSUE TITLE
[Issue] dubbo2.7.3升级到2.7.5报IllegalStateException: Invalid name="org.apache.dubbo.config.ProtocolConfig#0" contains illegal character，服务启动失败

### ISSUE THREAD

**Issue by mercyblitz** [apache/dubbo#5945]
Issue source :  https://github.com/apache/dubbo-spring-boot-project/issues/683

**Comment by huadong** [comment]
在springboot的配置文件application.properties加入:
>spring.application.name=dubboProvider

服务器启动不报错。

**Comment by mercyblitz** [comment]
> 服务器启动不报错。

Yes, it can be fixed as you said, However, `dubbo.application.name` or relative property has to configure in Spring scenarios.

**Comment by huadong** [comment]
当我打包到docker里面运行，虽然在application.properties进行了指定，还是会出现错误：
java.lang.IllegalStateException: Invalid name="org.apache.dubbo.config.ProtocolConfig#0"

**Comment by huadong** [comment]
![image](https://user-images.githubusercontent.com/2927763/78021840-7ee57400-7386-11ea-8aa6-942a9b41d076.png)

**Comment by huadong** [comment]
> > 服务器启动不报错。
> 
> Yes, it can be fixed as you said, However, `dubbo.application.name` or correlative property has to configure in Spring scenarios.

确实是很不方便。继续往`application.properties`加入：
>dubbo.protocol.name=dubbo

看来debug和打包在jar的环境下，配置文件加载是不一样的。

**Comment by zhangjun050754** [comment]
<img width="395" alt="360截图1771040411695141" src="https://user-images.githubusercontent.com/22231085/102524517-16852100-40d4-11eb-9194-d28b7b7b242d.png">

出现这个问题 怀疑是springboot加载配置文件顺序的问题 Spring Boot run as bootstrap.properties -> bootstrap.yml -> application.properties -> application.yml    duboo配置放在application中解决问题
